src/bank.rs

```diff
@@ -2210,6 +2215,37 @@ mod tests {
         assert_eq!(bank.checkpoint_depth(), 0);
     }
 
+    #[test]
+    fn test_bank_checkpoint_zero_balance() {
+        let alice = Mint::new(1_000);
+        let bank = Bank::new(&alice);
+        let bob = Keypair::new();
+        let charlie = Keypair::new();
+
+        // bob should have 500
+        bank.transfer(500, &alice.keypair(), bob.pubkey(), alice.last_id())
+            .unwrap();
+        assert_eq!(bank.get_balance(&bob.pubkey()), 500);
+        assert_eq!(bank.checkpoint_depth(), 0);
+
+        bank.checkpoint();
+        assert_eq!(bank.checkpoint_depth(), 1);
+
+        // charlie should have 500, alice should have 0
+        bank.transfer(500, &alice.keypair(), charlie.pubkey(), alice.last_id())
+            .unwrap();
+        assert_eq!(bank.get_balance(&charlie.pubkey()), 500);
+        assert_eq!(bank.get_balance(&alice.pubkey()), 0);
+
+        let account = bank.get_account(&alice.pubkey()).unwrap();
+        let default_account = Account::default();
+        assert_eq!(account.tokens, default_account.tokens);
```

Comment:
these assertions should also apply to a brand new bank, too, right?

can you compare bank.get_account() and Default::default() directly, instead of comparing each field?

Reply:
Yes, assertions should apply to fresh bank accounts.
I tried comparing get_account() to default(). It's not implemented.